mount/mount_linux.go

```diff
@@ -249,26 +334,30 @@ func parseMountOptions(options []string) (int, []string, bool) {
 		// If the option does not exist in the flags table or the flag
 		// is not supported on the platform,
 		// then it is a data value for a specific fs type
-		if f, exists := flags[o]; exists && f.flag != 0 {
+		if f, exists := flagsMap[o]; exists && f.flag != 0 {
 			if f.clear {
-				flag &^= f.flag
+				opt.flags &^= f.flag
 			} else {
-				flag |= f.flag
+				opt.flags |= f.flag
 			}
 		} else if o == loopOpt {
-			losetup = true
+			opt.losetup = true
+		} else if strings.HasPrefix(o, "uidmap=") {
+			opt.uidmap = strings.TrimPrefix(o, "uidmap=")
+		} else if strings.HasPrefix(o, "gidmap=") {
+			opt.gidmap = strings.TrimPrefix(o, "gidmap=")
 		} else {
 			data = append(data, o)
 		}
 	}
-	return flag, data, losetup
+	opt.data = strings.Join(data, ",")
+	return
 }
 
-func hasDirectIO(opts []string) (bool, []string) {
-	for idx, opt := range opts {
-		if opt == "direct-io" {
-			return true, append(opts[:idx], opts[idx+1:]...)
-		}
+func hasDirectIO(opts string) (bool, string) {
+	re := regexp.MustCompile(`direct-io,?`)
```

Comment:
I don't want to use regrex here, because checking option one by one should be correct than assuming `,` in the pattern.

Reply:
Done